tests/src/python/test_qgsrulebasedrenderer.py

```diff
@@ -0,0 +1,133 @@
+"""QGIS Unit tests for QgsRuleBasedRenderer
```

Comment:
I'm surprised this python file was added in Nov _2024_ :wink:

Reply:
Ah, we had one previously but because of a sip issue it randomly crashed. So all the tests got moved to c++. But now that sip issue is fixed 😁